### DISCUSSION TITLE
Is it possible to inject environment variables into the running xonsh from a Python subprocess?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4644] [General]
When I do data exploration work in xonsh (yes, in doesn't always have to be Jupyter!), I use some Python CLI programs which analyse files in the current directory and they create some output files.

After the Python program has run I would love to have the names of the output files avaible as environment variables of the calling xonsh. So is it possible to inject environment variables into the running xonsh from a Python subprocess?

**Comment by jnoortheen** [comment]
I think some kind of inter-process communication is needed. Something like custom signals/sockets/os.pipe can be used

**Comment by halloleo** [comment]
Cool, thanks, that's an idea! (But looks like *a lot of* work...)

I guess it might be a bit easier to span a new xonsh shell with the needed environment variables from the CLI program.
